core/domain/exp_jobs_one_off.py

```diff
@@ -720,3 +720,40 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class TranslatorToVoiceArtistOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating translator_ids to voice_artist_ids."""
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationRightsModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        translator_ids = item.translator_ids
+        if len(translator_ids) > 0:
+            item.voice_artist_ids = translator_ids
+            item.translator_ids = []
+            commit_message = 'Migrate from translator to voice artist'
+            commit_cmds = [{
+                'cmd': 'change_role',
+                'assignee_ids': translator_ids,
+                'old_role': 'translator',
+                'new_role': 'voice artist'
+            }]
+            item.commit('Admin', commit_message, commit_cmds)
+
+            exp_summary_model = exp_models.ExpSummaryModel.get_by_id(item.id)
+            if exp_summary_model is None:
+                return
```

Comment:
For an exploration having `ExplorationRightsModel` but no SummaryModel are we going to do a partial one-off without any yield? 
Also, do we need to check whether the model is deleted, like the one you have done for `ExplorationRightsModel`?

Reply:
If the RightsModel is not deleted, will SummaryModel be deleted?

Reply:
Done.